### PR TITLE
Refuel - change nozzle position calc

### LINKED ISSUE #5143: UAV Yabhon refueling issue
**Arma 3 Version:** `1.68` (stable)
**CBA Version:** `3.3.1` (stable)
**ACE3 Version:** `3.9.1` (stable)
 
**Mods:**
```
- CBA_A3
- ace
```

**Description:**
- Cannot connect the fuel nozzle with the Yabhon. 

**Steps to reproduce:**
- Place a fuel truck and a Yabhon UAV with bingo fuel on map. try to refuel the drone.

**Where did the issue occur?**
- Dedicated / Editor (Singleplayer)

**Placed Modules:**
- None

**RPT log file:**
- not needed

### PR DESCRIPTION
**Pull request by PabstMirror**
Fix #5143

Removes the old binary search and now uses `lineIntersectsSurfaces`
Default is to aim at player's viewing angle (what you see is what you get) instead of using model center

### CODE REVIEW

File: addons/refuel/functions/fnc_checkFuel.sqf [hunk 1]
```diff
@@ -21,7 +21,7 @@ params [["_unit", objNull, [objNull]], ["_target", objNull, [objNull]]];
 private _fuel = [_target] call FUNC(getFuel);
 
 [
-    REFUEL_PROGRESS_DURATION * 2.5,
+    TIME_PROGRESSBAR(REFUEL_PROGRESS_DURATION * 2),
```
Comment (Neviothr) [inline addons/refuel/functions/fnc_checkFuel.sqf]: Shouldn't `REFUEL_PROGRESS_DURATION` be a `GVAR()`?

File: addons/refuel/script_component.hpp [hunk 2]
```diff
@@ -29,3 +30,9 @@
     _weaponSelect = _unit getVariable QGVAR(selectedWeaponOnRefuel); \
     _unit selectWeapon _weaponSelect; \
     _unit setVariable [QGVAR(selectedWeaponOnRefuel), nil];
+
+#ifdef FAST_PROGRESSBARS
+    #define TIME_PROGRESSBAR(X) ((X) * 0.075)
+#else
+    #define TIME_PROGRESSBAR(X) (X)
+#endif
```
Comment (jonpas) [inline addons/refuel/script_component.hpp]: We could do that globally, and just set it to 0 or 0.5 (so it's visible that it happens at least) when defined?

File: addons/refuel/functions/fnc_checkFuel.sqf (hunk 1 above)
Comment (jonpas) [inline addons/refuel/functions/fnc_checkFuel.sqf]: No, this is not flow rate, it's how long it takes to check current fuel level of a vehicle. Macro is 2s, so this makes it 4s.

File: addons/refuel/script_component.hpp (hunk 2 above)
Comment (PabstMirror) [inline addons/refuel/script_component.hpp]: `TIME_PROGRESSBAR` is already in rearm
Rearm, Refuel are the main ones with some long progress bars that make testing tedious.

Might actually cause problems on some things like repacking and medical surgery, not sure.

Comment (jonpas) [inline addons/refuel/script_component.hpp]: I see. Would be done separate of this PR anyways.

**Pull request merged**